### PR TITLE
[Form] Change FormTypeGuesserChain to accept Traversable

### PR DESCRIPTION
**Pull request by enumag**
| Q | A |
| --- | --- |
| Branch? | master |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

I'm using symfony/form without the rest of the framework and this will make things a bit easier for me.

### CODE REVIEW

File: src/Symfony/Component/Form/FormTypeGuesserChain.php [hunk 1]
```diff
@@ -25,11 +25,15 @@ class FormTypeGuesserChain implements FormTypeGuesserInterface
...
+    public function __construct($guessers)
     {
+        if (!is_array($guessers) && !$guessers instanceof \Traversable) {
+            throw new UnexpectedTypeException($guessers, 'array or Traversable');
+        }
+
         foreach ($guessers as $guesser) {
             if (!$guesser instanceof FormTypeGuesserInterface) {
-                throw new UnexpectedTypeException($guesser, 'Symfony\Component\Form\FormTypeGuesserInterface');
+                throw new UnexpectedTypeException($guesser, FormTypeGuesserInterface::class);
```
Comment (fabpot) [inline src/Symfony/Component/Form/FormTypeGuesserChain.php]: This should be reverted

Comment (enumag) [inline src/Symfony/Component/Form/FormTypeGuesserChain.php]: It seemed logical but if you say so... reverted.

Comment (wouterj) [inline src/Symfony/Component/Form/FormTypeGuesserChain.php]: Problem is that's causing merge conflicts when merging older branches in the newer onces.

Comment (enumag) [inline src/Symfony/Component/Form/FormTypeGuesserChain.php]: Oh. That explains why Symfony still uses the old array syntax too. Thank you!

**Comment by enumag** [comment]
@fabpot Anything else to do here?

**Pull request merged**